cores/esp8266/WString.cpp

```diff
@@ -617,19 +617,35 @@ int String::indexOf(char ch, unsigned int fromIndex) const {
     return temp - buffer();
 }
 
-int String::indexOf(const String &s2) const {
+int String::indexOf(const __FlashStringHelper *s2) const {
     return indexOf(s2, 0);
 }
 
-int String::indexOf(const String &s2, unsigned int fromIndex) const {
+int String::indexOf(const __FlashStringHelper *s2, unsigned int fromIndex) const {
+    return indexOf((char*) s2, fromIndex);
```

Comment:
This will generate a warning when used because of the cast of a `const <blah>` to a mutable `<blah>`.

I'd recommend:  `return indexOf((const char*) s2, fromIndex);`